components/ontology-access/api/src/main/java/org/phenotips/ontology/internal/solr/AbstractHGNCSolrOntologyService.java

```diff
@@ -0,0 +1,339 @@
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.phenotips.ontology.internal.solr;
+
+import org.phenotips.obo2solr.ParameterPreparer;
+import org.phenotips.obo2solr.TermData;
+import org.phenotips.ontology.OntologyTerm;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+
+/**
+ * Ontologies processed from HGNC file share much of the processing code.
+ *
+ * @since 1.2RC1
+ * @version $Id$
+ */
+public abstract class AbstractHGNCSolrOntologyService extends AbstractSolrOntologyService
+{
+    /**
+     * The name of the Alternative ID field, used for older aliases of updated HPO terms.
+     */
+    protected static final String ALTERNATIVE_ID_FIELD_NAME = "alt_id";
+
+    protected static final String VERSION_FIELD_NAME = "version";
+
+    protected static final String SIZE_FIELD_NAME = "size";
+    protected static final String ROWS_FIELD_NAME = "rows";
+
+    private static final String FIELD_VALUE_SEPARATOR = "\\t";
+
+    private String infoServiceURL = "http://rest.genenames.org/info";
```

Comment:
As per my comment above, everything specific to HGNC should move to the class extending this one.